src/core/security/credentials.c

```diff
@@ -871,9 +871,12 @@ static void fake_oauth2_get_request_metadata(grpc_credentials *creds,
   grpc_fake_oauth2_credentials *c = (grpc_fake_oauth2_credentials *)creds;
 
   if (c->is_async) {
-    grpc_iomgr_add_callback(
-        on_simulated_token_fetch_done,
+    grpc_iomgr_closure *on_simulated_token_fetch_done_closure =
+        gpr_malloc(sizeof(grpc_iomgr_closure));
+    grpc_iomgr_managed_closure_init(
```

Comment:
This remains the single usage of the managed_closure concept. Course of action TBD, added TODO.